ssl/statem/statem_clnt.c

```diff
@@ -1136,6 +1136,21 @@ MSG_PROCESS_RETURN tls_process_server_hello(SSL *s, PACKET *pkt)
                session_id_len);
     }
 
+    if (!ssl_verify_client_session_version(s)) {
```

Comment:
Yeah, `s->hit` is not a guarantee here. I think this is the most safe way to do it atm.